src/Development/Shake/Command.hs

```diff
@@ -143,14 +143,19 @@ commandExplicit funcName copts results exe args = do
             Just LintTracker -> winTracker act
             Just LintFSATrace -> fsatrace act
             _ -> act exe args
+        track (rs,ws) = do
+          cwd <- liftIO $ getCurrentDirectory
+          let blacklisted = shakeLintFilter opts
+              ham = filter blacklisted
+              rel = map (makeRelative cwd)
+          trackRead $ rel $ ham rs
+          trackWrite $ rel $ ham ws
 
         winTracker act = do
             (dir, cleanup) <- liftIO newTempDir
             flip actionFinally cleanup $ do
                 res <- act "tracker" $ "/if":dir:"/c":exe:args
-                (rs, ws) <- liftIO $ trackerFiles dir
-                trackRead rs
-                trackWrite ws
+                liftIO (trackerFiles dir) >>= track
```

Comment:
I'm not sure about this... Does tracker.exe report absolute paths? If not, this should probably be piped through `makeAbsolute` prior to calling `track`.